### PR TITLE
UMUS-28 search snippet highlighting + UMUS-102 no wildcard queries

### PR DESCRIPTION
**Pull request by jesconstantine**
## Tickets
* [UMUS-28](https://palantir.atlassian.net/browse/UMUS-28)
* [UMUS-102](https://palantir.atlassian.net/browse/UMUS-102)

## Description 
This PR updates our fork of the solr-faceted-search-react codebase to:
1. Enable a `mainQueryField` flag to be set, allowing an existing search field to be used to populate the main query param ("q") instead of being added as a filter list field ("fl")
1. Enable empty search results (vs always sending wildcard queries) by adding a `buildMainQuery` method which populates the "q" param based on the presence and value of the `mainQueryField` flag
1. Enable search result snippet highlighting by adding a `buildHighlight` method which populates enables solr highlighting and, when passed, adds subsequent `hl` params
1. Add and update any necessary unit tests

## To test
* See functional test script in https://github.com/palantirnet/umus/compare/umus-28-result-snippet-highlight to verify this work
* Optional: you can clone this repo, check out this branch, and run `npm run test` to verify passing unit tests

### CODE REVIEW

**Comment by byrond** [comment]
@jesconstantine I looked over the code, checked out this branch locally and did `npm run test`.  It sounds like when I tested https://github.com/palantirnet/umus/pull/32, that confirms functionality of this PR.  Do I understand that correctly?

**Comment by jesconstantine** [comment]
@byrond yep, you got it!  the functional test for this happens in https://github.com/palantirnet/umus/pull/32 -- thanks so much!  Happy to let this wait for merging until @froboy has taken a look himself too.

File: src/api/solr-query.js [hunk 1]
```diff
@@ -51,7 +51,9 @@ const fieldToQueryFilter = (field) => {
 	return null;
 };
 
-const buildQuery = (fields) => fields
+const buildQuery = (fields, mainQueryField) => fields
+	// Do not include main query field in filter field query param.
```
Comment (froboy) [inline src/api/solr-query.js:55]: There's some mixing of spaces and tabs here, but I think I'll resolve that in a different PR, as there's probably a lot to reconcile. I think the old project was using tabs, but React seems to go with spaces https://github.com/facebook/react/blob/master/.editorconfig.

Comment (jesconstantine) [reply src/api/solr-query.js:55]: I'm happy to convert tabs to spaces for any file that's touched as part of this PR for now, if you'd like?

**Comment by froboy** [comment]
@jesconstantine nice work. I gave the code a brief review too and everything looks like it's nicely abstracted I'd like to get some words into the readme about implementing highlighting, but I'll leave that as a todo for later. Also very nice work on keeping the tests up to date... I've never even looked at those kind of tests, but it all seems reasonable.

**Pull request merged**